### PR TITLE
Power of 2

### PR DESCRIPTION
**Pull request by utkarsh006**
## Related Issue
- Power of 2 #1351

## Proposed Changes
- I have added the following things: -
- Problem Statement
- Examples
- Algorithm
- Code
- Time Complexity & Space Complexity

## Additional Info
[LINK OF THE PROBLEM STATEMENT](https://leetcode.com/problems/power-of-two/)

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

We were supposed to find if a number can be represented as power of two.

I have added the Problem Statement, Examples, an Algorithm on how to approach this problem, a commented Code, and written the Time Complexity and Space Complexity. And finally updated the README.md.

This change helps to solve a very famous problem of Recursion i.e. Power of Two and this problem helps in understanding the concept of Recursion in depth.

I coded this problem in VS Code and tested the given problem on the test cases in the range: -231 <= n <= 231 - 1. And it is working fine for them.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Input : 16  | Input : 5   |
|image|image|

### CODE REVIEW

**Comment by Mrjoy832** [comment]
Add one index in the readme top kindly as there are many numbers of questions

**Comment by utkarsh006** [comment]
> Add one index in the readme top kindly as there are many numbers of questions

I have done the required changes

**Review by Mrjoy832 — APPROVED** [review]

File: R-Recursion/Power_of_two.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* TIME COMPLEXITY : O(N)
+SPACE COMPLEXITY : O(N)   */
```
Comment (Lakhankumawat) [inline R-Recursion/Power_of_two.cpp:2]: File name must be camel cased `PowerOfTwo.cpp`

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: R-Recursion/Power_of_two.cpp (hunk 1 above)
Comment (utkarsh006) [reply R-Recursion/Power_of_two.cpp]: > File name must be camel cased `PowerOfTwo.cpp`

Done the changes kindly review @Lakhankumawat

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

**Comment by Lakhankumawat** [comment]
Contiuous Good Contribution -1 [ CGC ]
mention this pr number in your next pr and if i found it eligible and there are some efforts in the pr i will increase the level of that pr.

**Pull request merged**